src/Storages/RabbitMQ/ReadBufferFromRabbitMQConsumer.cpp

```diff
@@ -408,10 +277,10 @@ bool ReadBufferFromRabbitMQConsumer::nextImpl()
     if (stopped || !allowed)
         return false;
 
-    if (messages.tryPop(current))
+    if (received.tryPop(current))
     {
-        auto * new_position = const_cast<char *>(current.data());
-        BufferBase::set(new_position, current.size(), 0);
+        auto * new_position = const_cast<char *>(current.message.data());
+        BufferBase::set(new_position, current.message.size(), 0);
```

Comment:
Are you sure about consistency between other parts of `current`? It's not obvious from code.

Reply:
yes, why would it if fields of `MessageData` are never changed once the message is pushed to queue and then when `current` is updated there is `buffer->allowNext()`, before which other parts of `current` are collected and only after that `current` will be updated again

Reply:
by the way, there is a test that covers it, so all ok